### ISSUE TITLE
Why there is a variation for three different runs using same setting?

### ISSUE THREAD

**Issue by lrbrrr** [ultralytics/yolov5#2216]
## ❔Question
![Screenshot from 2021-02-15 12-06-00](https://user-images.githubusercontent.com/78729788/107913381-5a868c00-6f86-11eb-8660-a3312f8bf934.png)

Between three different trials,
Dataset: same
hyperparameters: same
number of epochs: same
dimension of image: same
epoch size: same

Why do I see a variation in the cls_loss which eventually also computes different mAP? 

Anyone out there please guide.

**Comment by glenn-jocher** [comment]
@lrbrrr not sure. May have to do with your dataset, as I've not observed anything similar on COCO.

**Comment by lrbrrr** [comment]
I found this unusual too. I've used the same dataset across trials. The box_loss and obj_loss gets an overlap but not the cls_loss. I wonder what could be the reason.

I read about situation where variation could be seen. One of the many says it is dependent on computing resources and I've been doing this on google colab. Could it be the reason the resources allocation is different each time?

**Comment by glenn-jocher** [comment]
@lrbrrr yeah, it's a pretty significant difference in cls_loss. The other two variations between runs look much more normal to me. In COCO differences across runs are maybe <1% 1-sigma across all metrics and losses, consistent with your last two losses. Not sure what's happening to your cls loss. The only advice I have are all the default suggestions, i.e. restart your machine, create a new python 3.8 environment, clone a fresh repo and install dependencies and go from there.

The GCP VMs used in Colab do vary by region and resources/GPUs etc, but no reason I know of that cls should be differently affected by any external factor compared to box and obj.

**Comment by glenn-jocher** [comment]
@lrbrrr I tried to reproduce with COCO128 in a colab notebook, but I got consistent results as expected.

The command I used was:
```python
# Train YOLOv5s on COCO128 for 3 epochs
for _ in range(3):
  !python train.py --img 640 --batch 16 --epochs 300 --data coco128.yaml --weights yolov5s.pt --nosave --cache
```

## STEP x axis
<img width="1000" alt="Screen Shot 2021-02-15 at 10 14 01 PM" src="https://user-images.githubusercontent.com/26833433/108025400-4751db80-6fdb-11eb-88b1-735b8d206dfe.png">

## WALL CLOCK x axis
<img width="1000" alt="Screen Shot 2021-02-15 at 10 14 15 PM" src="https://user-images.githubusercontent.com/26833433/108025462-5cc70580-6fdb-11eb-833c-5b78628712b1.png">
